src/main/java/com/wavefront/sdk/common/metrics/WavefrontSdkMetricsRegistry.java

```diff
@@ -159,6 +160,15 @@ public void run() {
             Number value = (Number) ((WavefrontSdkGauge) metric).getValue();
             wavefrontMetricSender.sendMetric(name, value.doubleValue(), timestamp, source, tags);
           }
+        }  else if (metric instanceof WavefrontSdkDeltaCounter) {
+          if (!name.startsWith(Constants.DELTA_PREFIX) && !name.startsWith(Constants.DELTA_PREFIX_2)) {
+            name = Constants.DELTA_PREFIX + name;
+          }
```

Comment:
why can't we do this when registering the metric into the `metrics` CHM?

Reply:
`metrics` do not have full name, so we add the Delta to the name before sending.